### PR TITLE
Standardization fixes

### PR DESCRIPTION
**Pull request by Jakz**
This pull request is aimed to refactor and beautify the code:

- removed non-standard constructs allowed by GCC compiler but not by C++ standard (eg. designated initializers, `uint`)
- increased utilization of some C++11/C++17 STL functions like `std::clamp`, `std::stoi` and such
- `std::chrono` instead of `<sys/time.h>`
- continued replacing POSIX like file system access functions with `std::filesystem`
- added build options to force windowed mode and specific resolution for debugging

### CODE REVIEW

File: src/gmenu2x.cpp [hunk 1]
```diff
@@ -62,19 +62,17 @@
...
 #include <sys/types.h>
-#include <dirent.h>
 
 //for soundcard
-#include <sys/ioctl.h>
+/*#include <sys/ioctl.h>
 #include <linux/soundcard.h>
 
-#include <sys/mman.h>
+#include <sys/mman.h>*/
```
Comment (pcercuei) [inline src/gmenu2x.cpp]: If these are useless just remove them.

File: src/gmenu2x.cpp [hunk 2]
```diff
@@ -130,6 +128,8 @@ static void set_handler(int signal, void (*handler)(int))
 	sig.sa_handler = handler;
 	sig.sa_flags |= SA_RESTART;
 	sigaction(signal, &sig, NULL);
+
+
```
Comment (pcercuei) [inline src/gmenu2x.cpp]: No need for these.

File: src/gmenu2x.cpp [hunk 3]
```diff
@@ -194,7 +194,7 @@ GMenu2X::GMenu2X() : input(*this), sc(this)
 	 * https://github.com/mthuurne/opendingux-buildroot/blob
 	 * 			/opendingux-2010.11/package/sdl/sdl-fbcon-clear-onexit.patch
 	 */
-	setenv("SDL_FBCON_DONT_CLEAR", "1", 0);
+	setenv("SDL_FBCON_DONT_CLEAR", "1", false);
```
Comment (pcercuei) [inline src/gmenu2x.cpp]: setenv() is a C function, the 3rd parameter is an ```int```, so I think 0 is clearer.

**Review by pcercuei — CHANGES_REQUESTED** [review]

**Comment by Jakz** [comment]
👍

**Pull request merged**